src/expression.d

```diff
@@ -11069,6 +11069,18 @@ extern (C++) final class DeleteExp : UnaExp
                 return new ErrorExp();
         }
 
+        // unsafe
+        if (!sc.intypeof && sc.func)
+        {
+            if (sc.func.setUnsafe())
+            {
+                error("%s is not @safe but is used in @safe function %s", toChars(), sc.func.toChars());
+                err = true;
+            }
+        }
+        if (err)
```

Comment:
I don't think that works out too well with the `err |= ...` used earlier. Also, frequent `new ErrorExp` makes for bloated code.